docs/framework/react/guide/tanstack-start.md

```diff
@@ -0,0 +1,246 @@
+
+We'll then update our `package.json` to reference the new Vinxi entry point:
+
+```json
+{
+  "//": "...",
+  "scripts": {
+    "dev": "vinxi dev",
+    "build": "vinxi build",
+    "start": "vinxi start"
+  }
+}
+```
+
+To tell Vinxi that it should start TanStack Start's minimal behavior, we need to configure the `app.config.ts` file:
+
+```typescript
+// app.config.ts
+import { defineConfig } from '@tanstack/start/config'
+
+export default defineConfig({});
+```
+
+# Add the Basic Templating
+
+There are four required files for TanStack Start usage:
+
+1) The router configuration
+2) The server entry point
+3) The client entry point
+4) The root of your application
+
+Once configuration is done, we'll have a file tree that looks like the following:
+
+```
+.
+├── app/
+│   ├── routes/
+│   │   └── `__root.tsx`
+│   ├── `client.tsx`
+│   ├── `router.tsx`
+│   ├── `routeTree.gen.ts`
+│   └── `ssr.tsx`
+├── `.gitignore`
+├── `app.config.ts`
+├── `package.json`
+└── `tsconfig.json`
+```
+
+## The Router Configuration
+
+This is the file that will dictate the behavior of TanStack Router used within Start. Here, you can configure everything
+from the default [preloading functionality](./preloading.md) to [caching staleness](./data-loading.md).
+
+```tsx
+// app/router.tsx
+import { createRouter as createTanStackRouter } from '@tanstack/react-router'
+import { routeTree } from './routeTree.gen'
+
+export function createRouter() {
+  const router = createTanStackRouter({
+    routeTree
+  })
+
+  return router
+}
+
+declare module '@tanstack/react-router' {
+  interface Register {
+    router: ReturnType<typeof createRouter>
+  }
+}
+```
+
+> `routeTree.gen.ts` is not a file you're expected to have at this point.
+> It will be generated when you run TanStack Start (via `npm run dev` or `npm run start`) for the first time.
+
+## The Server Entry Point
+
+As TanStack Start is an [SSR](https://unicorn-utterances.com/posts/what-is-ssr-and-ssg) framework, we need to pipe this router
+information to our server entry point:
+
+```tsx
+// app/server.tsx
+import {
+  createStartHandler,
+  defaultStreamHandler,
+} from '@tanstack/start/server'
+import { getRouterManifest } from '@tanstack/start/router-manifest'
+
+import { createRouter } from './router'
+
+export default createStartHandler({
+  createRouter,
+  getRouterManifest,
+})(defaultStreamHandler)
+```
+
+This allows us to know what routes and loaders we need to execute when the user hits a given route.
+
+## The Client Entry Point
+
+Now we need a way to hydrate our client-side JavaScript once the route resolves to the client. We do this by piping the same
+router information to our client entry point:
+
+```tsx
+// app/client.tsx
+import { hydrateRoot } from 'react-dom/client'
+import { StartClient } from '@tanstack/start'
+import { createRouter } from './router'
+
+const router = createRouter()
+
+hydrateRoot(document.getElementById('root')!, <StartClient router={router} />)
```

Comment:
When I `vinxi build` this file as-written in my app, I get the following line in the build logs:
```
virtual:$vinxi/handler/client (1:97): "default" is not exported by "app/client.tsx", imported by "virtual:$vinxi/handler/client".
```

<img width="917" alt="image" src="https://github.com/TanStack/router/assets/721128/241caa3c-c6b7-43fe-a784-887bbc7a9c0f">

Reply:
Yeah, I get this error too; but this is unrelated to the quickstart guide.